runtime/compiler/net/StreamExceptions.hpp

```diff
@@ -37,27 +37,47 @@ class StreamFailure: public virtual std::exception
    std::string _message;
    };
 
-class StreamCancel: public virtual std::exception
+class StreamInterrupted: public virtual std::exception
    {
 public:
-   StreamCancel(MessageType type, uint64_t clientId=0) : _type(type), _clientId(clientId) { }
+   StreamInterrupted() : _message("Compilation interrupted at JITClient's request") { }
    virtual const char* what() const throw()
       {
-      if (_type == MessageType::clientTerminate)
-         return "Client session terminated at client's request";
-      else
-         return "Compilation cancelled by client";
+      return _message.c_str();
+      }
+private:
+   std::string _message;
+   };
+
+class StreamConnectionTerminate: public virtual std::exception
+   {
+public:
+   StreamConnectionTerminate() : _message("Connection terminated at JITClient's request") { }
+   virtual const char* what() const throw()
+      {
+      return _message.c_str();
       }
-   MessageType getType() const
+private:
+   std::string _message;
+   };
+
+class StreamClientSessionTerminate: public virtual std::exception
+   {
+public:
+   StreamClientSessionTerminate(uint64_t clientId) : _clientId(clientId)
       {
-      return _type;
+      _message = "JITClient session " + std::to_string(_clientId) + " terminated at JITClient's request";
```

Comment:
Similar as above, we don't need to store a message because we have all the information we need to print it in the `what()` routine.